rtt_rosparam/src/rtt_rosparam_service.cpp

```diff
@@ -12,6 +12,29 @@
 
 #include <ros/ros.h>
 
+#ifndef DECLARE_ROSPARAM_GETTER_SETTER
+#define DECLARE_ROSPARAM_GETTER_SETTER(function_name,return_type) \
```

Comment:
Should we also add all the `*Relative`, `*Absolute`, `*Private`, ... variants for each type-specific getter/setter operation? Currently they would default to policy `Relative`. For the case a component wants to get/set a parameter in a component-private namespace, the only possible invocation is

```cpp
rosparam.getString("~name_of_the_component/parameter", s)
rosparam.getString("~" + getName() + "/parameter", s)
```

and `getName()` is not one of the default operations for components. In comparison:

```cpp
rosparam.getStringComponentPrivate("parameter", s)
```

I actually would like the terms "Local" and "Global" more instead of "Relative" and "Absolute" (for another PR).